packages/components/src/DateField/DateInput.tsx

```diff
@@ -0,0 +1,54 @@
+import { ReactElement } from 'react';
+import { DateInput, Group } from 'react-aria-components';
+import type RAC from 'react-aria-components';
+
+import { cn, useClassNames } from '@marigold/system';
+
+import { DateSegment } from './DateSegment';
+
+type RemovedProps = 'style' | 'className' | 'children';
+
+export interface DateInputProps extends Omit<RAC.DateInputProps, RemovedProps> {
+  variant?: string;
+  size?: string;
+  action?: ReactElement;
+}
+
+const _DateInput = ({ variant, size, action, ...props }: DateInputProps) => {
+  const classNames = useClassNames({ component: 'DateField', variant, size });
+
+  return (
+    <Group className={cn('flex flex-1 items-center', classNames.field)}>
+      <DateInput className={cn('flex flex-1 items-center')} {...props}>
+        {segment => (
+          <DateSegment
+            className={cn(
+              'group/segment',
+              'text-center leading-none outline-0',
+              '[data-type="literal]"' && 'p-[0.75px]',
+              classNames.segment
+            )}
+            segment={segment}
+          />
+        )}
+      </DateInput>
+      {action ? (
```

Comment:
Should we make this not part of the date input? It is only needed by the picker anyway. Yes, it should be removed from the DateField 👀

Reply:
the only child of a `DateInput` which is allowed is a `segment`, so I move it as it is to the `DateInput`, can you be more speciic? thank youu 👀  Also I leave the icon on the `DateField` because I don't want to change the design. 😬